tests/test_talonsrx.py

```diff
@@ -485,15 +463,181 @@ def test_basemotorcontroller_selectedSensorPosition(talon, cdata):
     assert talon.getSelectedSensorPosition(1) == 0
 
 
-@pytest.mark.xfail(raises=NotImplementedError)
-def test_basemotorcontroller_setSensorPhase(talon):
+def test_basemotorcontroller_setSensorPhase(talon, cdata):
     talon.setSensorPhase(True)
+    assert cdata['sensor_phase'] == True
 
 
-@pytest.mark.xfail(raises=NotImplementedError)
 def test_basemotorcontroller_setStatusFramePeriod(talon):
     talon.setStatusFramePeriod(1, 2, 3)
 
 
+def test_basemotorcontroller_getClosedLoopTarget(talon, cdata):
+    cdata['pid0_target'] = 44
+    cdata['pid1_target'] = 55
+    assert talon.getClosedLoopTarget(0) == 44
+    assert talon.getClosedLoopTarget(1) == 55
+
+
 def test_basemotorcontroller_valueUpdated(talon):
     talon.valueUpdated()
+
+
+def assert_is_almost_equal(expected, actual, rel_tol=0, abs_tol=0.00001):
+    assert math.isclose(expected, actual, rel_tol=rel_tol, abs_tol=abs_tol), "expected: %f, got: %f" % (expected, actual)
```

Comment:
I still think this assert should be moved to each test function. pytest will do introspection for you, so no need for the message, or the mess of specifying your own `rel_tol` and `abs_tol` that don't match what you actually want.